velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -2837,3 +2837,31 @@ TEST_F(DateTimeFunctionsTest, dateParse) {
   VELOX_ASSERT_THROW(
       dateParse("116", "%y+"), "Invalid format: \"116\" is malformed at \"6\"");
 }
+
+TEST_F(DateTimeFunctionsTest, timeZoneHour) {
+  const auto timezone_hour = [&](const char* time, const char* timezone) {
+    Timestamp ts = util::fromTimestampString(time);
+    auto timestamp = ts.toMillis();
+    auto hour = evaluateWithTimestampWithTimezone<int64_t>(
+                    "timezone_hour(c0)", timestamp, timezone)
+                    .value();
+    return hour;
+  };
+
+  // Asia/Kolkata - should return 5 throughout the year
+  EXPECT_EQ(5, timezone_hour("2023-01-01 03:20:00", "Asia/Kolkata"));
+  EXPECT_EQ(5, timezone_hour("2023-06-01 03:20:00", "Asia/Kolkata"));
+  // America/Los_Angeles - Day light savings starts from March 12
+  EXPECT_EQ(-8, timezone_hour("2023-03-11 12:00:00", "America/Los_Angeles"));
```

Comment:
What do we get for `timezone_hour("2023-03-12 02:30:00", "America/Los_Angeles")` and `timezone_hour("2023-11-15 01:30:00", "America/Los_Angeles")`?

Reply:
@Yuhta Am getting the following values.
timezone_hour("2023-03-12 02:30:00", "America/Los_Angeles") -> -8
timezone_hour("2023-11-15 01:30:00", "America/Los_Angeles") -> -8
Do I need to add these 2 also in test case?

Reply:
Can you check these 2 are compatible with Presto Java behavior?  Once confirmed we can add these 2 to unit test.

Reply:
@Yuhta Confirmed with Java behavior.
<img width="827" alt="image" src="https://user-images.githubusercontent.com/111376031/233535939-9850286b-a4b5-45e7-bd63-656e70817a7c.png">

Added these 2 to unit test.

Reply:
Oops, I think I made a typo, the second one should be 2023-11-05 01:30:00, can you check this one?

Once confirmed this should be good.  There are also some merge conflicts to resolve, then we can merge this.

Reply:
@Yuhta For 2023-11-05 01:30:00 am getting -7, which is the same in Java too.
<img width="831" alt="image" src="https://user-images.githubusercontent.com/111376031/233697938-28a5dea6-96d5-4461-bbf9-937f62897df5.png">

Added that case also in unit test.